include/functional/functor.hpp

```diff
@@ -45,13 +38,8 @@ template <typename Functor, typename... Args> struct functor final {
     requires std::same_as<std::remove_cvref_t<decltype(self)>, functor>
              && monadic_invocable<functor_type, decltype(v), Args...>
   {
-    return
-        [&v, &self]<unsigned... I>(
-            std::integer_sequence<unsigned, I...>) noexcept -> decltype(auto) {
-          return Functor::apply::template operator()( //
-              std::forward<decltype(v)>(v),           //
-              detail::get<I>(std::forward<decltype(self)>(self).data)...);
-        }(std::make_integer_sequence<unsigned, size>{});
+    constexpr static typename functor_type::apply fn{};
+    return data_t::invoke(FWD(self).data, FWD(fn), FWD(v));
```

Comment:
this peeves me off - would be so nice if we inlined not_tuple into here and just... not have two more inlining contexts...

future improvement.